tools/backups2datalad/datasetter.py

```diff
@@ -216,13 +236,20 @@ def get_dandiset_stats(self, ds: Dataset) -> DandisetStats:
                 dandiset_metadata_file,
             ):
                 if filestat["type"] == "dataset":
-                    substat = self.get_dandiset_stats(Dataset(filestat["path"]))
-                    files += substat.files
-                    size += substat.size
+                    zarr_ds = Dataset(filestat["path"])
+                    zarr_id = Path(self.get_remote_url(ds)).name
+                    try:
+                        zarr_stat = substats[zarr_id]
+                    except KeyError:
+                        zarr_stat, subsubstats = self.get_dandiset_stats(zarr_ds)
+                        assert not subsubstats
+                        substats[zarr_id] = zarr_stat
+                    files += zarr_stat.files
+                    size += zarr_stat.size
                 else:
                     files += 1
                     size += filestat["bytesize"]
-        return DandisetStats(files=files, size=size)
+        return (DandisetStats(files=files, size=size), substats)
```

Comment:
would be nice to address #153 while at it to ensure that now more elaborate implementation of this function works correctly.